cypress/e2e/API/customers.cy.js

```diff
@@ -0,0 +1,167 @@
+/// <reference types="Cypress" />
+
+describe('GET /customers', () => {
+    const CUSTOMERS_API_URL = `${Cypress.env('API_URL')}`
+    it('Validar a lista de clientes com sucesso', () => {
+      cy.request(
+        'GET',
+        `${CUSTOMERS_API_URL}customers?page=1&limit=10&size=All`
+      ).then(({ status }) => {
+        expect(status).to.eq(200)
+      })
+    })
+  
+    it('Validar filtro pelo tamanho da empresa', () => {
+      const tamanhos = [
+        'Small',
+        'Medium',
+        'Enterprise',
+        'Large Enterprise',
+        'Very Large Enterprise',
+      ]
+  
+      tamanhos.forEach((tamanho) => {
+        cy.request(
+          'GET',
+          `${CUSTOMERS_API_URL}customers?page=1&limit=10&size=${tamanho}`
+        ).then(({ body }) => {
+          body.customers.forEach((customer) => {
+            expect(customer.size).to.eq(tamanho)
+          })
+        })
+      })
+    })
+  
+    it('Validar filtro por paginação', () => {
+      const qtdePaginas = ['5', '10', '20', '50']
+      const totalDePagPorLimite = [10, 5, 3, 1]
+  
+      qtdePaginas.forEach((limite, indice) => {
+        cy.request(
+          'GET',
+          `${CUSTOMERS_API_URL}customers?page=1&limit=${limite}&size=All`
+        ).then(({ body }) => {
+          expect(body.customers).to.have.length(limite)
+          expect(body.pageInfo.totalPages).to.eq(totalDePagPorLimite[indice])
+        })
+      })
+    })
+  
+    it('Valider as propriedades customer e pagInfo da resposta', () => {
+      cy.request(
+        'GET',
+        `${CUSTOMERS_API_URL}customers?page=1&limit=5&size=All`
+      ).then(({ body }) => {
+        expect(body).to.have.property('customers')
+        expect(body).to.have.property('pageInfo')
+        expect(body.customers).to.be.an('array')
+        expect(body.customers.length).to.be.greaterThan(0)
+  
+        body.customers.forEach((customer) => {
+          expect(customer).to.have.property('id')
+          expect(customer).to.have.property('name')
+          expect(customer).to.have.property('employees')
+          expect(customer).to.have.property('size')
+          expect(customer).to.have.property('address')
+          expect(customer).to.have.property('contactInfo')
+        })
+        expect(body.pageInfo).to.have.property('currentPage')
+        expect(body.pageInfo).to.have.property('totalPages')
+        expect(body.pageInfo).to.have.property('totalCustomers')
+      })
+    })
+  
+    context('Validar solicitações inválidas', () => {
+      it('Por filtrar página negativa', () => {
+        cy.request({
+          method: 'GET',
+          url: `${CUSTOMERS_API_URL}customers?page=-3&limit=5&size=Large%20Enterprise`,
```

Comment:
Visto que o propósito do teste é filtar por `page`, deixo os outros _query params_ para os _defaults_ da API.

```suggestion
          url: `${CUSTOMERS_API_URL}customers?page=-3`,
```

Reply:
O mesmo vale para os outros testes de cenários negativos.